### ISSUE TITLE
Cannot complete correctly for pytorch

### ISSUE THREAD

**Issue by sillybun** [davidhalter/jedi#1316]
```
source = "import torch\ntorch.ma"
script = jedi.Script(source, 2, len("torch.ma"), "e.py")
script.completions()
```
And i get:
`[<Completion: manager_path>, <Completion: manual_seed>, <Completion: math>]`

BUT, what I what is `matmul`. Why cannot jedi complete this function?

Besides, jedi cannot complete `torch.from_numpy` as well.

**Comment by sillybun** [comment]
I find adding `'torch'` to `jedi.settings.auto_import_modules` solves this problem. It there any better solutions?

**Comment by davidhalter** [comment]
I feel like Jedi has a really hard time finding matmul. By reading the pytorch code I have trouble to find it. Can you enlighten me where it is defined?

**Comment by sillybun** [comment]
`matmul` appears in `__init__pyi` file:
```
__init__.pyi:    def matmul(self, other: 'Tensor', *, out: Optional['Tensor']=None) -> 'Tensor': ...
__init__.pyi:def matmul(self: Tensor, other: Tensor, *, out: Optional[Tensor]=None) -> Tensor: ...
onnx/symbolic.py:def matmul(g, self, other):
```
Since `matmul` is written in c++, it's hard to find its definition.

**Comment by davidhalter** [comment]
Oh, now I see. Please wait for #839 and tell me if it's not working once that is finished.


Note to myself: The stubs are here: https://github.com/pytorch/pytorch/blob/master/torch/__init__.pyi.in

**Comment by Coderx7** [comment]
@davidhalter Hi, I faced the same issue with Pytorch, and created a new issue in the vscode section :  
I don't know if we should continue this here or there, since on one hand this is related to Jedi, and on the other hand this is happening in vscode (Python extension related which uses Jedi).  
Anyway here is the ticket on the vscode : https://github.com/microsoft/vscode-python/issues/5643

**Comment by davidhalter** [comment]
@Coderx7 Happy to have to conversation here. However it's now just a matter of time until we have a release. On the master branch you should be able to work with pytorch stubs.

Feel free to try :) You just need to clone Jedi, `git submodule update --init` and `pip install -e .`. There's probably still a few issues. I know of about 3 for now, but I'm fixing them in the next few days and after that the master branch should be better than what it was before.

I'm actively looking for feedback for this branch, so you're the perfect "customer" :).

**Comment by Coderx7** [comment]
@davidhalter , seems they have it baked into the Python extension and there is no separate folder for Jedi! 
Seems we cant just upgrade to the newer version manually! :(

**Comment by davidhalter** [comment]
Why? The Jedi files need to be somewhere ;-)

**Comment by Coderx7** [comment]
I found it. it was located in the `C:\Users\Your_User_Name\.vscode\extensions\ms-python.python-2019.4.12954\pythonFiles\lib\python\jedi`
I have an older version of Jedi (0.9 I guess) in my Anaconda3 distribution (which is the default Python package in my system) and the version shipped with VSCode is 0.12.0 as it reads in the `__init__.py`.
So how do I go about it now? will just replacing the old files with the newer ones from the master branch do it?

**Comment by davidhalter** [comment]
0.9 is really old. That's probably one of the first "ok" versions, but please use a newer one. No idea why they ship such old versions. Please use the latest version. Switching from 0.12.0 to 0.13.2 should probably be worry-free, however switching from 0.9 might cause some problems (some things have been deprecated and removed since then).

**Comment by Coderx7** [comment]
I updated my anaconda's Jedi to the latest version. I also downloaded the latest master branch, and copied the Jedi subfolder to the VSCode Extensions folder, replacing the old version there. After doing so, the intellisense broke and didnt work anymore, I had to revert back my changes (i.e using the former 0.12.0 version in VSCode directory) . 
Was this all I needed to do? or did I miss something?

**Comment by davidhalter** [comment]
For VSCode, I'm really unsure. I would probably need an exception to tell what's going on. There's probably logs somewhere or a debug function that you can enable.

**Comment by Coderx7** [comment]
Here is [the log](https://gist.githubusercontent.com/Coderx7/6ef0664687f394cf3a366d0c025d44be/raw/271252e6539c33127d98a0a3123257dce684df7d/vscode_console_log.log) concerning my latest changes (updating the Jedi to the latest version) .
